Themis/Services/Submission/TextSubmissionServiceImpl.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  TextSubmissionServiceImpl.swift
+//  Themis
+//
+//  Created by Tarlan Ismayilsoy on 23.05.23.
+//
+
+import Foundation
+import SharedModels
+import APIClient
+import DesignLibrary
+import Common
+
+class TextSubmissionServiceImpl: SubmissionService {
+    
+    typealias SubmissionType = TextSubmission
+    
+    let client = APIClient()
+    
+    // MARK: - Get All Submissions
+    private struct GetAllSubmissionsRequest: APIRequest {
+        typealias Response = [Submission]
+        
+        var exerciseId: Int
+        
+        var method: HTTPMethod {
+            .get
+        }
+        
+        var resourceName: String {
+            "api/exercises/\(exerciseId)/text-submissions"
+        }
+    }
+    
+    func getAllSubmissions(exerciseId: Int) async throws -> [Submission] {
+        try await client.sendRequest(GetAllSubmissionsRequest(exerciseId: exerciseId)).get().0
+    }
+    
+    // MARK: - Get Tutor Submissions
+    private struct GetTutorSubmissionsRequest: APIRequest {
```

Comment:
"Tutor submissions" are submissions which have been graded by the tutor in the past, right? Maybe we could have a comment explaining this somewhere?